src/network/protocol.rs

```diff
@@ -86,7 +86,7 @@ impl InputBytes {
         for p in 0..num_players {
             let start = p * size;
             let end = start + size;
-            let input = *bytemuck::try_from_bytes::<T::Input>(&self.bytes[start..end])
+            let input = *bytemuck::checked::try_from_bytes::<T::Input>(&self.bytes[start..end])
```

Comment:
That's an important change.

Reply:
Can you elaborate what is actually checked here?

Reply:
Sure, first, we need to understand the behaviour before this change:

With the restriction "AnyBitValid", enums could not work because an enum cannot have "any" bits ; mainly because its type flag, but probably other types are concerned (references, bool...)

Nothing is really checked if the type is "AnyBitPattern" though

the "checked" prefixed leads to: 
https://github.com/Lokathor/bytemuck/blob/6b1b7cecd32902b240a8e8593534600a09269d14/src/checked.rs#L209 ; where there is additional checks in case there are not valid bit patterns 😅.

All "AnyBitPatterns" implement "CheckedBitPattern" though, so no additional checks are performed for most types :).